File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.py

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        if not root:
            return -1
        hp = []
        heapq.heapify(hp)

        def DFS(curNode: Optional[TreeNode]):
            if not curNode:
                return
            heapq.heappush(hp, -curNode.val)
            if len(hp) > k:
                heapq.heappop(hp)
            if curNode.left:
                DFS(curNode.left)
            if curNode.right:
                DFS(curNode.right)

        DFS(root)

        return -hp[0] if hp else -1

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> ans;
        return helper(root ,ans )

        
    }
    vector<int> helper(TreenNode * curNode , ans vector<int>){
        if(curNdoe == nullptr){
            return ans;
        }

        helper(curNode->right , ans.push_back(curNode->right));
    }
};
